### PR TITLE
Upate qchem op generators to return qml.Hamiltonian

### PR DESCRIPTION
**Pull request by lillian542**
**Context:**
Updates the qchem generators to return as either a `qml.ops.Hamiltonian` or `qml.ops.LinearCombination` (depending on whether op_math is enabled) as part of standardizing generators and keeping them consistent with the `op_math` setting.

**Description of the Change:**
These ones already returned a new opmath operator (SProd) if `active_new_opmath` and a legacy operator (Hamiltonian) otherwise, because they are created using dunder methods rather than explicit constructors. This change makes it so they will return `LinearCombination` instead, because this is more analogous to `Hamiltonian`, and some of the generator behaviour depends on that.

Added tests for both old and new opmath checking the generator return type, and tagged the parameter frequency tests (functionality dependent on `op.generator()`) to run with both.

**Benefits:**
- The qchem op generators are consistently either "`qml.Hamiltonian`" (whatever that may mean in the current context) or `SparseHamiltonian`. 
- We can remove one of our `xfail` marks in the optimize folder that was a result of the discrepancy here
- We can revert a test change that we missed previously that represented a real change in functionality and *should* have been an `xfail`

### CODE REVIEW

File: tests/test_operation.py [hunk 1]
```diff
@@ -2554,19 +2555,13 @@ def test_has_grad_method(self):
         assert qml.operation.has_grad_method(self.rot)
         assert not qml.operation.has_grad_method(self.cnot)
 
-    @pytest.mark.usefixtures("use_legacy_opmath")
-    def test_gen_is_multi_term_hamiltonian_legacy_opmath(self):
+    def test_gen_is_multi_term_hamiltonian(self):
```
Comment (lillian542) [inline tests/test_operation.py:2558]: The `gen_is_multi_term_hamiltonian` function is used in decomposition/tape expansion logic, and is not just a type check. This function should not have been updated to run with `use_legacy_opmath`; its failure indicates a genuine problem (related to the QNG test failure in optimize), and not just a difference in convention.

**Comment by lillian542** [comment]
[sc-57982]

File: doc/releases/changelog-dev.md [hunk 2]
```diff
@@ -139,9 +139,10 @@
 * The `molecular_hamiltonian` function calls `PySCF` directly when `method='pyscf'` is selected.
   [(#5118)](https://github.com/PennyLaneAI/pennylane/pull/5118)
 
-* All generators in the source code (except those in the `qchem` module) no longer return
-  `Hamiltonian` or `Tensor` instances. Wherever possible, these return `Sum`, `SProd`, and `Prod` instances.
+* The generators in the source code return operators consistent with the global setting for 
+  `qml.operator.active_new_opmath()` whereever possible.
```
Comment (PietropaoloFrisoni) [inline doc/releases/changelog-dev.md:143]:
```suggestion
  `qml.operator.active_new_opmath()` wherever possible.
```

File: tests/test_operation.py (hunk 1 above)
Comment (PietropaoloFrisoni) [reply tests/test_operation.py:2558]: Yes, it has been modified recently with temporary fixes to enable opmath globally, so I'm not surprised that we missed this

**Review by PietropaoloFrisoni — CHANGES_REQUESTED** [review]
Just a typo in the changelog. Nice!

**Review by PietropaoloFrisoni — APPROVED** [review]
Cool stuff

**Review by mudit2812 — APPROVED** [review]

**Pull request merged**